MINIBALL: Supplement the Miniball introductory doc

yaoyinnan opened the pull request:
Supplement the Miniball introductory documentation, including design, architecture, steps to start the guest, getting started, and supported platforms.

ASOC: https://github.com/openanolis/dragonball-sandbox/issues/155

Signed-off-by: yaoyinnan <yaoyinnan@foxmail.com>

studychao commented:
Thanks for your doc:)
I'll read that tomorrow~

studychao commented on crates/dbs-miniball/README.md:
```diff
@@ -1,21 +1,528 @@
+`aarch64-unknown-linux-gnu` and `aarch64-unknown-linux-musl` toolchains,
+defaulting (probably) to `aarch64-unknown-linux-gnu` on ARM, because it's also
+*Tier 1 supported* since Rust 1.49.
 
```
If we are inspired by rust-vmm's vmm-reference repo, please add an acknowledge part for their work:)

yaoyinnan replied:
Updated it.

studychao commented on crates/dbs-miniball/README.md:
````diff
@@ -1,21 +1,528 @@
+event_manager.add_subscriber(wrapped_exit_handler.0.clone());
+```
+
+4. Configure the vCPUs. This is done through `vm-vcpu` crate, which for now is a
````
Actually, lots of functions here could be accomplished by `dbs-boot`, `dbs-arch`

studychao commented on crates/dbs-miniball/README.md:
```diff
@@ -1,21 +1,528 @@
+
+
+
+5. Configure legacy devices. This is done partially through `kvm-ioctls`,
```
You could take a look at [dbs-legacy-devices](https://github.com/openanolis/dragonball-sandbox/tree/main/crates/dbs-legacy-devices)

studychao commented on crates/dbs-miniball/README.md:
````diff
@@ -1,21 +1,528 @@
+vmm.add_i8042_device()?;
+```
+
+6. Configure root block device. This is done through `vm-virtio`. Device event
````
[dbs-virtio-devices](https://github.com/openanolis/dragonball-sandbox/tree/main/crates/dbs-virtio-devices) could be useful .

studychao commented on crates/dbs-miniball/README.md:
```diff
@@ -1,21 +1,528 @@
+users can disregard the `vmm` crate altogether and build their own, referring
+to it only for guidelines.
+
+![Overview](docs/overview.png)
```
Could you add the repo from dragonball-sandbox here?

studychao reviewed (commented):
Hi, Yinnan.
Thanks for your work, please take a look at my comments.

yaoyinnan commented:
I refer to [vmm-reference](https://github.com/rust-vmm/vmm-reference) to sort out the code of the guest startup process. As the project progresses, I will use the crates of `dragonball-sandbox` to replace the crates of `rust-vmm` mentioned in the document.

yaoyinnan committed:
MINIBALL: Supplement the Miniball introductory doc

Supplement the Miniball introductory documentation, including design, architecture, steps to start the guest, getting started, and supported platforms.

Signed-off-by: yaoyinnan <yaoyinnan@foxmail.com>
